### PR TITLE
Revise ZU.cleanISBN

### PR DESCRIPTION
**Pull request by aurimasv**
Fixes regression from 37921b09100cd35697173be39f272837ec7c1f95 but also this should be better overall

### CODE REVIEW

**Comment by simonster** [comment]
This would be an excellent candidate for unit testing.

**Comment by aurimasv** [comment]
Tests added for `cleanISSN`, `cleanISBN`, and `toISBN13`

**Comment by aurimasv** [comment]
Note that there's a fairly exhaustive test on what characters should and should not be ignored between ISSN/ISBN digits. It goes through all the characters in the BMP, so that takes a while (~1-1.5 seconds). We probably don't have to be that exhaustive though, so I can trim it down if there's a concern.

File: test/tests/utilities.js [hunk 1]
```diff
@@ -17,4 +17,159 @@ describe("Zotero.Utilities", function() {
 			}
 		});
 	});
+	describe("cleanISBN", function() {
+		let cleanISBN = Zotero.Utilities.cleanISBN;
+		it("should return false for non-ISBN string", function() {
+			assert.isFalse(cleanISBN(''), 'returned false for empty string');
+			assert.isFalse(cleanISBN('Random String 123'), 'retruned false for non-ISBN string');
```
Comment (zuphilip) [inline test/tests/utilities.js]: Typo (retruned) which occurs in two other lines as well.

Comment (aurimasv) [inline test/tests/utilities.js]: Thanks! I fixed this and squashed.

**Comment by aurimasv** [comment]
@simonster can you take a look and let me know if this is how I should proceed with tests for other components

**Comment by simonster** [comment]
@dstillman has more experience with mocha/chai than I do, but this looks good to me. I think a simpler test ignored chars would have been sufficient, although testing all of them doesn't hurt and 3 seconds isn't a big deal.

**Comment by simonster** [comment]
Thanks!

**Comment by dstillman** [comment]
We're getting a Travis failure due to a timeout for that test.

```
x  should strip off internal characters in ISSN string

Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test
```
We could [adjust the timeout](http://mochajs.org/#test-specific-timeouts) for that test, though I agree with Simon that a simpler test would probably be sufficient here.

**Comment by dstillman** [comment]
(It's true that 3 seconds isn't a big deal with 30 tests, but we might feel differently when we have hundreds.)

**Pull request merged**